frontend/src/components/design-system/input/ButtonComponent.vue

```diff
@@ -0,0 +1,270 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+import ButtonText from '@/components/design-system/typography/ButtonText.vue';
+
+export interface Props {
+  tag?: 'button' | 'a';
+  href?: string;
+  kind: 'link' | 'outlined' | 'filled' | 'answer';
+  size?: 'medium' | 'small';
+  color?: 'neutral' | 'primary' | 'secondary';
+  selected?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  tag: 'button',
+  href: undefined,
+  size: 'medium',
+  color: 'neutral',
+  selected: false,
+});
+
+const classes = computed(() => ({
+  [`button--${props.kind}`]: props.kind,
+  [`button--${props.size}`]: props.size,
+  [`button--${props.color}`]: props.color,
+  [`button--${props.selected}`]: props.selected,
+}));
+
+const computedTag = computed(() => (props.href ? 'a' : props.tag));
+
+const computedSize = computed(() =>
+  props.kind === 'link' ? props.size : 'medium'
+);
+</script>
+
+<template>
+  <component :is="computedTag" :href="href" :class="['button', classes]">
+    <span
+      v-if="$slots.icon"
+      :class="['icon', $slots.default ? 'icon--before' : 'icon--only']"
+    >
+      <slot name="icon" :size="computedSize" />
+    </span>
+    <ButtonText v-if="$slots.default" :size="computedSize">
+      <slot />
+    </ButtonText>
+    <span v-if="$slots.iconAfter" :class="['icon', 'icon--after']">
+      <slot name="iconAfter" :size="computedSize" />
```

Comment:
Řešil jsi případ, kdy je tlačítko přes celou šířku a text a ikona jsou každý na jedné straně?

Reply:
Ne, ale to nikde není, nebo jsem to aspoň nikde neviděl.

Reply:
No tam je právě přesný opak, ne? Že je tam mezi mezera, to je výchozí chování (`space-between`).